libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -33,20 +35,22 @@ Compass::compass_cal_update()
     if (is_calibrating()) {
         _cal_has_run = true;
         return;
-    } else if (_cal_has_run && auto_reboot()) {
+    } else if (_cal_has_run && _auto_reboot()) {
         hal.scheduler->delay(1000);
         hal.scheduler->reboot(false);
     }
 }
 
 bool
-Compass::start_calibration(uint8_t i, bool retry, bool autosave, float delay, bool autoreboot)
+Compass::_start_calibration(uint8_t i, bool retry, float delay)
 {
     if (!healthy(i)) {
         return false;
     }
-    memset(_reports_sent,0,sizeof(_reports_sent));
-    if (!is_calibrating() && delay > 0.5f) {
+    if (!use_for_yaw(i)) {
+        return false;
```

Comment:
except that puts the burden on the GCS developer and complicates UIs
For the (rare!) people that want to calibrate a disabled compass they can change COMPASS_USE*, calibrate, then disable again.
We should make the common case as simple as possible, and keep the UIs as simple in the GCS as we can. Not calibrating disabled compasses seems to achieve that.